prefix-map.c

```diff
@@ -35,40 +35,40 @@ static void init_prefix_map(struct prefix_map *prefix_map,
 static void add_prefix_item(struct prefix_map *prefix_map,
 			    struct prefix_item *item)
 {
-	struct prefix_map_entry *e = xmalloc(sizeof(*e)), *e2;
+	struct prefix_map_entry e = { { NULL } }, *e2;
 	int j;
 
-	e->item = item;
-	e->name = e->item->name;
+	e.item = item;
+	e.name = item->name;
```

Comment:
Maybe add:

If (strlen(e.name) < prefix_map->min_length) BUG(...);

Reply:
(I’m viewing this on my iPad so I can’t tell if this case is handled elsewhere.)

Reply:
It is not, but I do not want to overdo the checks for invariants...